packages/core-data/src/selectors.ts

```diff
@@ -891,6 +891,57 @@ export const getEditedEntityRecord = createSelector(
 	}
 );
 
+/**
+ * Returns a list of entity records, merged with their edits.
+ *
+ * @param state     State tree.
+ * @param kind      Entity kind.
+ * @param name      Entity name.
+ * @param recordIds Record IDs.
+ *
+ * @return The list of entity records, merged with their edits.
+ */
+export const getEditedEntityRecords = createSelector(
+	< EntityRecord extends ET.EntityRecord< any > >(
+		state: State,
+		kind: string,
+		name: string,
+		recordIds: EntityRecordKey[]
+	): Array< ET.Updatable< EntityRecord > | false > => {
+		return recordIds.map( ( recordId ) =>
+			getEditedEntityRecord( state, kind, name, recordId )
```

Comment:
> Is it because we are not triggering a finishResolution for each record as part of that call?

That's probably true.

What are our options? Should we use `getEntityRecords` instead of `getEntityRecord` in `getEditedEntityRecords`

Reply:
Actually I stand corrected, the `finishResolution` does work correctly. The requests I am seeing are `OPTIONS` calls triggered by the `canUser` selector for getting the permissions: https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/resolvers.js#L524-L533

Reply:
Can run permissions selector after items are resolved. This way we can avoid dispatching a permission request for each item.

Reply:
It should. The `getEntityRecords` selector will try to resolve permissions for each entity when possible (ref #64504). However, we had a race condition because it called both selectors simultaneously.

> Ok, so the problem was that I was passing the postIds in as strings into the permission call which re-triggered the requests. 

Maybe we should normalize IDs before passing them to selectors.

Reply:
> Maybe we should normalize IDs before passing them to selectors.

Yeah not a bad plan, with the current changes I suppose we are doing a form of normalizing by relying on the ID type that the entity record has.